src/components/Markings/components/ShapeEditorSvg.js

```diff
@@ -165,9 +164,14 @@ class ShapeEditorSvg extends Component {
                             displayToNativeRatioY={this.props.displayToNativeRatioY}
                             showCorners={this.props.mode === 'edit' && selectedShape}
                             isDeletable={this.props.mode === 'erase'}
-                            ref={ref => this.setState({
-                                shapeRefs: R.set(R.lensProp(index), ref ,this.state.shapeRefs)
-                            })}
+                            ref={ref => {
+                                if (ref) {
+                                    this.setState(currentState => {
+                                        currentState.shapeRefs = R.set(R.lensProp(index), ref, currentState.shapeRefs)
+                                        return currentState
```

Comment:
What is the purpose of this change?

Reply:
There was a race condition in the code because this code was being called multiple times so close together it was essentially concurrent. React optimizes `setState` by batching updates and committing them at the same time. However, if you pass in a function to `setState` it will run the updates in sequence, thus preventing race conditions on the `state.shapeRefs` object.

Explained in: 
https://stackoverflow.com/questions/30338577/reactjs-concurrent-setstate-race-condition